examples/03_connectivity/plot_confounds_removal.py

```diff
@@ -0,0 +1,170 @@
+# Friston 24 motion parameters
+# -------------------------------------
+# They include the 6 motion parameters of the current volume
+# and the preceding volume, plus each of these values squared.
+
+# Read the 6 motion parameters from the confounds file
+import numpy as np
+csv_confounds = np.recfromcsv(adhd.confounds[0], delimiter='\t',
+                              usecols=('motionx', 'motiony', 'motionz',
+                                       'motionpitch', 'motionroll',
+                                       'motionyaw'))
+current_volume_motion = csv_confounds.view(dtype=float).reshape(-1, 6)
+previous_volume_motion = np.vstack((current_volume_motion[0],
+                                    current_volume_motion[:-1]))
+confounds = np.hstack((current_volume_motion, previous_volume_motion,
+                       current_volume_motion ** 2,
+                       previous_volume_motion ** 2))
+
+##########################################################################
+# Anatomical CompCor
+# ----------------------------------------------
+# Signals within noisy ROIs, such as WM and CSF, are assumed to reflect noise.
+# The anatomical CompCor approach removes the first principal components
+# of noisy ROIs.
+#
+# First, we define the tissue masks.
+# As we do not have an anatomical image, we are relying on group-level masks.
+# Note that subject-specific masks would give more accurate results.
+
+##########################################################################
+# We give the locations of probabilistic group masks
+icbm152 = datasets.fetch_icbm152_2009()
+mask_images = {'gm': icbm152.gm, 'wm': icbm152.wm, 'csf': icbm152.csf}
+
+##########################################################################
+# and use them to compute binary masks.
+from nilearn import image
+for tissue_name, threshold in zip(['wm', 'gm', 'csf'], ['.9', '.5', '.3']):
+    mask_images[tissue_name] = image.math_img('img > ' + threshold,
+                                              img=mask_images[tissue_name])
+
+##########################################################################
+# We erode WM and CSF masks, to avoid including any signal of neuronal origin.
+from scipy import ndimage
+for tissue_name in ['wm', 'csf']:
+    mask_img = mask_images[tissue_name]
+    eroded_mask_data = ndimage.binary_erosion(mask_img.get_data(),
+                                              iterations=2)
+    mask_images[tissue_name] = image.new_img_like(mask_img, eroded_mask_data)
+
+##########################################################################
+# Next, we resample the tissue masks to the functional image
+from nilearn import image, masking
+mean_func_img = image.mean_img(func_filename)
+func_mask_img = masking.compute_epi_mask(mean_func_img)
+affine = mean_func_img.get_affine()
+shape = mean_func_img.get_data().shape
+for tissue_name, mask_img in mask_images.items():
+    mask_images[tissue_name] = image.resample_img(mask_img, target_shape=shape,
```

Comment:
May be you can have a look at ```resample_to_img``` function to avoid some lines such as affine, shape, etc. source image to target image. Note that default is continuous interpolation in that function